src/sentry/testutils/cases.py

```diff
@@ -106,6 +106,38 @@ def setup_dummy_auth_provider(self):
     def tasks(self):
         return TaskRunner()
 
+    @classmethod
+    @contextmanager
+    def static_asset_manifest(cls, manifest_data):
+        dist_path = "src/sentry/static/sentry/dist"
+        manifest_path = f"{dist_path}/manifest.json"
+
+        with open(manifest_path, "w") as manifest_fp:
+            manifest_fp.truncate(0)
+            json.dump(manifest_data, manifest_fp)
+
+        files = []
+        for file_path in manifest_data.values():
+            full_path = f"{dist_path}/{file_path}"
+            # make directories in case they don't exist
+            # (e.g. dist path should exist, but subdirs won't)
+            os.makedirs(os.path.dirname(full_path), exist_ok=True)
+            open(full_path, "a").close()
```

Comment:
Ideally we should start adopting pathlib, this would be more readable with `touch`. I'm fine with this though.